src/http_util.rs

```diff
@@ -26,6 +26,37 @@ pub fn get_client() -> Client<Connector, hyper::Body> {
   Client::builder().build(c)
 }
 
+// Construct the next uri based on base uri and location header fragment
+// https://tools.ietf.org/html/rfc3986#section-4.2
+fn maybe_uri_from_location(base_uri: &Uri, location: &str) -> Option<Uri> {
```

Comment:
1) I think it's better to call this `resolve_uri_from_location`
2) Can you use triple slashes for the comments, so it shows up in rustdoc?
3) Also please describe in the docs when the "None" case occurs.

Reply:
(We haven't been doing triple slash comments for Rust - but I will be trying to fix that in the future - as we move towards releasing a crate)

Reply:
Done. Removed the `Option` wrapper since it is actually not needed with current design.